storage/src/mmr/verification.rs

```diff
@@ -370,6 +372,81 @@ fn peak_hash_from_range<'a, H: CHasher>(
     Ok(hasher.node_hash(pos, &left_hash.unwrap(), &right_hash.unwrap()))
 }
 
+/// A proof that a key has a particular (currently active) value in a [MutableMmr] with a particular
+/// root hash.
+pub struct KeyValueProof<K: Array, V: Array, H: CHasher> {
+    /// The offset of the leaf storing the currently active value for the key in the update tree.
+    pub tree_pos: u64,
+
+    /// A proof that the record at the given offset has a certain key and value.
+    pub updates_proof: Proof<H>,
+
+    /// A proof that the value is currently active for the key.
+    pub is_active_proof: Proof<H>,
+
+    /// The bitmap chunk containing the is_active bit of the referenced update record.
+    pub state_bitmap_chunk: H::Digest,
+
+    pub phantom_key: PhantomData<K>,
```

Comment:
Recommend making these non-public and adding a `_` prefix (the linter isn't warning because you made them pub).

Reply:
Added constructor and changed fields to private w/ _ prefix.